api/dp-openapi.yaml

```diff
@@ -784,6 +1132,163 @@ components:
         state:
           type: string
 
+    wordcheck_data:
+      type: object
+      properties:
+        langsWithDict:
+          type: array
+          items:
+            type: string
+        messages:
+          type: object
+        puncChars:
+          type: string
+
+    proof_page:
+      type: object
+      properties:
+        pagename:
+          type: string
+        image_url:
+          type: string
+        text:
+          type: string
+
+    page_text:
+      type: object
+      properties:
+        text:
+          type: string
+
+    round_data:
+      type: object
+      properties:
+        id:
+          type: string
+          example: F1
+        name:
+          type: string
+          description: the translated description
+          example: Formatting Round 1
+        formatting:
+          type: boolean
+          description: true if it is a formatting round
+          example: true
+        document:
+          type: string
+          description: url of the guidelines for the round
+          example: https://www.pgdp.net/wiki/DP_Official_Documentation:Formatting/Formatting_Guidelines
+
+    page_check:
+      description: check text using given languages and ad hoc good words
+      type: object
+      properties:
+        text:
+          type: string
+        auxlanguage:
+          type: string
+        adhocgoodwords:
+          type: array
+          items:
+            type: string
+
+    page_save:
+      type: object
+      properties:
+        new_page_state:
+          type: string
+          description: saved or temp
+          example: saved
+        text:
+          type: string
+          description: text to save
+
+    page_return:
+      type: object
+      properties:
+        new_page_state:
+          type: string
+          description: avail
+
+    page_mark_bad:
+      type: object
+      properties:
+        new_page_state:
+          type: string
+          example: bad
+        reason:
+          type: integer
+          description: index into array of reasons
+
+    pickersets:
+      type: array
+      items:
+        type: object
+        properties:
+          name:
+            type: string
+            example: basic-latin
+          subsets:
+            type: array
+            items:
+              type: object
+              properties:
+                name:
+                  type: string
+                  example: ˆ
+                title:
+                  type: string
+                  example: Letters with acute, grave, and circumflex
+                rows:
```

Comment:
Rows feels like a UI concept that doesn't belong in the REST api.

Reply:
Yes, I suppose it is only important for upper and lower case which we should describe more explicitly.